sys/unix/nledl.c

```diff
@@ -6,70 +6,102 @@
 
 #include "nledl.h"
 
-void
-nledl_init(nle_ctx_t *nledl, nle_obs *obs, nle_seeds_init_t *seed_init)
-{
-    nledl->dlhandle = dlopen(nledl->dlpath, RTLD_LAZY);
-
-    if (!nledl->dlhandle) {
-        fprintf(stderr, "%s\n", dlerror());
-        exit(EXIT_FAILURE);
-    }
-
-    dlerror(); /* Clear any existing error */
-
+#if defined(__linux__) && defined(__x86_64__)
+#define HASSHARED
```

Comment:
There is still a runtime option of using dlopen etc even if the shared mode is builtin. I'd like to leave this possibility open just in case it doesn't work on someones computer. I can imagine some version of gcc, linker etc to produce a shared object that my code will not link successfully.

Reply:
Ah yes that makes sense.